### PR TITLE
reader "back" button

### LINKED ISSUE #1198: Missing "back" button after following hyperlink
When one follows an hyperlink in an EPUB that takes you, for example, to the glossary at the end of the book, there is not "back" button to take you where you were before clicking on hyperlink. This is very annoying.

### PR DESCRIPTION
**Pull request by panaC**
fixes #1198

### CODE REVIEW

**Comment by panaC** [comment]
what is the state of this PR @danielweck @llemeurfr ?
can you merge it ? (after fixing conflicts)

**Review by danielweck — DISMISSED** [review]
Reader.tsx will probably conflict significantly with the PDF reader PR, which is more important. So let's wait.

File: src/resources/locales/en.json [hunk 1]
```diff
@@ -284,7 +284,8 @@
             "pdfscalemode": "Page fit or width",
             "quitFullscreenTitle": "quit fullscreen mode",
             "readBookTitle": "start reading the book",
-            "settingsTitle": "settings"
+            "settingsTitle": "settings",
+            "historyPrevious": "history previous"
```
Comment (llemeurfr) [inline src/resources/locales/en.json:288]: better label: "Back in navigation history".

More generally, please be careful to have a capital on the first letter of each label.

**Review by llemeurfr — DISMISSED** [review]
We now have two "back" buttons, not good for the UX. 
Proposal: change the button closing the reading view for a "cross" icon and change its label from "Back to the bookshelf" to "Close the publication".

**Comment by panaC** [comment]
Hello @llemeurfr, @gautierchomel,

Can you tell me in details how to implement this back button.

I understood that the back button go back in locator history to many times and "break" the experience. I can settle it to go to n-1 only. 
In this case when a user click on an internal link (footnote, section, ...) it can go back only one time.

Should i do implement this proposal ? https://github.com/edrlab/thorium-reader/pull/1200#pullrequestreview-1102003097

**Comment by danielweck** [comment]
> Proposal: change the button closing the reading view for a "cross" icon and change its label from "Back to the bookshelf" to "Close the publication".

Mhmm, I think changing the well-established "back arrow" icon button will break the user experience significantly. Also note that this button is not displayed when the library window is already open, typically when there are several opened reader windows (there is a separate icon to trigger this opt-in mode). In other words, I also think that changing the label to "close the publication" is a bad idea, as there is another mechanism in place to close the reader window (the "back arrow" actually means to go back to the bookshelf / library window).

**Comment by danielweck** [comment]
Do the navigation back/forward buttons (similar to web browser UI) need to be displayed in Thorium's top toolbar? Could they be located at the top of the "navigation" slide-out panel that displays the table of contents, etc.?
Note that the bottom-right corner already have left/right arrows, so the GUI is already cognitively loaded with "previous / next" metaphors.

**Review by gautierchomel — COMMENTED** [review]
Work as described. 

* I feel that we have to include click on link event to address the original issue (user story: _I clicked on a link inside the publication that as no backlink and I want to go back to original location_). 
* There is no visual indication about buttons state (active or inactive), they stay in grey with or without history element to go to. They should have (color + label) indicator of inactive state. (and I wonder if this should not be also the case for Left and Right button which are inactives at the start or at the end of the publication).

File: src/resources/locales/en.json (hunk 1 above)
Comment (danielweck) [reply src/resources/locales/en.json]: outdated, fixed since

**Review by danielweck — APPROVED** [review]

**Comment by danielweck** [comment]
TODO: keyboard shortcuts for navigation history back/forward, possibly only "back" if we decide to remove "forward".

**Comment by gautierchomel** [comment]
Works fine and now include click on link event and visual state (active / inactive). Very usefull new feature!

**Review by gautierchomel — APPROVED** [review]

**Comment by danielweck** [comment]
@gautierchomel what about?
Screenshot 2022-10-14 at 11 52 53


Screenshot 2022-10-14 at 11 52 42

**Comment by danielweck** [comment]
@panaC I tested `Computer-Systems_-A-Programmer's-Perspective.epub` and I am not able to reproduce any navigation / pagination problems that are specific to changes in this PR. Thorium from the `develop` branch exhibits similar occasional hiccups during the positioning of hyperlinks targets, but nothing major in my tests.

**Comment by danielweck** [comment]
@nvdaes FYI I added a new keyboard shortcut in the English Wiki page. The new "history navigation" feature has now been merged into the develop branch, but Thorium has not officially been published / updated yet (meanwhile, automated developer builds can be used for testing the new functionality, ahead of an official release announcement).

English wiki:
https://github.com/edrlab/thorium-reader/wiki/Keyboard-shortcuts

Spanish wiki:
https://github.com/edrlab/thorium-reader/wiki/Keyboard-shortcuts-(Spanish)

Text:
`CTRL` + `backspace` with `shift` modifier for the inverse operation (`NavigatePreviousHistory` / `NavigateNextHistory`): in reader window, go back or forward in the navigation history

@gautierchomel , something new for the Thorium documentation? https://github.com/edrlab/thorium-reader-doc

**Comment by gautierchomel** [comment]
Updated documentation ready to be pushed once this is released!

**Comment by nvdaes** [comment]
Thanks Daniel: I"ll take care of this tomorrow or on Sunday, translating the shortcut into Spanish and posting this to nvdaes.github.io, which I will update just while needed, that is, when a webpage by Edrlab is available in Spanish I"ll request people to go there.

Enviado desde mi iPhone

**Comment by nvdaes** [comment]
HelloDaniel, Gautier and all:

I've updated the [post about Thorium on my NVDA's spanish website](https://nvdaes.github.io/epub7).

Before audio commands, I've added the following line:

> * `CTRL` + `backspace` - control + retroceso, con modificador `shift` para la operación inversa (`NavigatePreviousHistory` / `NavigateNextHistory` - navegar al anterior elemento del historial / navegar al siguiente elemento del historial): en la ventana del lector, retrocede o avanza en el historial de navegación.

Feel free to update the spanish webpage on the wiki, and spanish speakers, please let me know if you detect any mistake or possible improvement on this page.

PS: FYI, I've linked Edrlab website on a [webpageto demonstrate WCAG 2.1 criteria to be evaluated with NVDA](https://nvdaes.github.io/nvdaes/weba11y), with other links for useful info about accessibility.

**Comment by danielweck** [comment]
Hello Noelia, I forgot to say thank you for the Spanish translation of the previous / next navigation history keyboard shortcut, which I added to the Thorium wiki (same bullet point location as your blog post):
* https://github.com/edrlab/thorium-reader/wiki/Keyboard-shortcuts-(Spanish)
* https://nvdaes.github.io/epub7

**Comment by nvdaes** [comment]
Thanks so much Daniel.

Enviado desde mi iPhone

**Pull request merged**